datafusion/src/physical_plan/metrics/value.rs

```diff
@@ -77,6 +77,62 @@ impl Count {
     }
 }
 
+/// A gauge is the simplest metrics type. It just returns a value.
+/// For example, you can easily expose current memory consumption with a gauge.
+///
+/// Note `clone`ing gauge update the same underlying metrics
+#[derive(Debug, Clone)]
+pub struct Gauge {
+    /// value of the metric counter
+    value: std::sync::Arc<AtomicUsize>,
+}
+
+impl PartialEq for Gauge {
+    fn eq(&self, other: &Self) -> bool {
+        self.value().eq(&other.value())
+    }
+}
+
+impl Display for Gauge {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.value())
+    }
+}
+
+impl Default for Gauge {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Gauge {
+    /// create a new gauge
+    pub fn new() -> Self {
+        Self {
+            value: Arc::new(AtomicUsize::new(0)),
+        }
+    }
+
+    /// Add `n` to the metric's value
+    pub fn add(&self, n: usize) {
```

Comment:
better to add `dec` function if we need dec the value.

Reply:
Currently, I haven't found a use case that needs to decrease the value of the gauge instead of setting the value to a new one. for example, during `spill` I set it to 0. therefore the API is not added for the moment.

Reply:
We discussed this previously in https://github.com/apache/arrow-datafusion/issues/1569. I will create a new issue later.

Reply:
If we don't use the `dec/sub`, we no need to add it.
Thanks @yjshen